h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -181,7 +187,9 @@ public DHistogram(String name, final int nbins, int nbins_cats, byte isInt, doub
       _step = 1.0f;                           // Fixed stepsize
     } else {
       _step = xbins / (maxEx - min);              // Step size for linear interpolation, using mul instead of div
-      assert _step > 0 && !Double.isInfinite(_step) : "Histogram step size for column '" + name + "' is invalid: " + _step + ".";
+      if(_step < 0 || Double.isInfinite(_step) || Double.isNaN(_step))
```

Comment:
Yeah, I suppose, the prior assert was there for step < 0 but <=0 makes more sense.